modules/elasticsearch/src/test/java/org/testcontainers/elasticsearch/ElasticsearchContainerTest.java

```diff
@@ -252,29 +258,71 @@ public void incompatibleSettingsTest() {
-
-            client = RestClient.builder(HttpHost.create("https://" + container.getHttpHostAddress()))
-                .setHttpClientConfigCallback(httpClientBuilder -> {
-                    httpClientBuilder.setDefaultCredentialsProvider(credentialsProvider);
-                    httpClientBuilder.setSSLContext(container.createSslContextFromCa());
-                    return httpClientBuilder;
-                })
-                .build();
+            Response response = getClusterHealth(container);
+            assertThat(response.getStatusLine().getStatusCode(), is(200));
+            assertThat(EntityUtils.toString(response.getEntity()), containsString("cluster_name"));
+        }
+    }
 
-            Response response = client.performRequest(new Request("GET", "/_cluster/health"));
+    @Test
+    public void testElasticsearch8SecureByDefaultCustomCaCertFails() throws Exception {
+        final MountableFile mountableFile = MountableFile.forClasspathResource("http_ca.crt");
+        String caPath = "/tmp/http_ca.crt";
+        try (ElasticsearchContainer container = new ElasticsearchContainer("docker.elastic.co/elasticsearch/elasticsearch:8.1.2")
+            .withCopyToContainer(mountableFile, caPath)
+            .withCertPath(caPath)) {
+
+            container.start();
+
+            // this is expected, as a different cert is used for creating the SSL context
+            assertThrows("PKIX path validation failed: java.security.cert.CertPathValidatorException: Path does not chain with any of the trust anchors", SSLHandshakeException.class, () -> getClusterHealth(container));
+        }
+    }
+
+    @Test
+    public void testElasticsearch8SecureByDefaultFailsSilentlyOnLatestImages() throws Exception {
+        // this test exists for custom images by users that use the `latest` tag
+        // even though the version might be older than version 8
+        // this tags an old 7.x version as :latest
+        tagImage("docker.elastic.co/elasticsearch/elasticsearch:7.9.2", "elasticsearch-tc-older-release", "latest");
+        DockerImageName image = DockerImageName.parse("elasticsearch-tc-older-release:latest").asCompatibleSubstituteFor("docker.elastic.co/elasticsearch/elasticsearch");
+
+        try (ElasticsearchContainer container = new ElasticsearchContainer(image)) {
+            container.start();
+
+            Response response = getClient(container).performRequest(new Request("GET", "/_cluster/health"));
             assertThat(response.getStatusLine().getStatusCode(), is(200));
             assertThat(EntityUtils.toString(response.getEntity()), containsString("cluster_name"));
         }
     }
 
+    private void tagImage(String sourceImage, String targetImage, String targetTag) throws InterruptedException {
+        DockerClient client = DockerClientFactory.instance().client();
+        client.pullImageCmd(sourceImage).exec(new PullImageResultCallback()).awaitCompletion();
```

Comment:
Please use `RemoteDockerImage` (something like `new RemoteDockerImage(DockerImageName.parse(sourceImage)).get()`), as it will take caching, auth and many other aspects into an account (vs raw pull)

Reply:
Here is an example in our code base:
https://github.com/testcontainers/testcontainers-java/blob/02782d9cb711baebaa504bc00669acdd84e10e51/modules/localstack/src/test/java/org/testcontainers/containers/localstack/LegacyModeTest.java#L74-L83

Reply:
done